### PR TITLE
Fix validation of select custom options

### PR DESCRIPTION
**Pull request by igrslv**

### CODE REVIEW

**Comment by vkublytskyi** [comment]
@igrslv, thank you for your contribution. Please accept the contributors license agreement so this PR can be further processed. This can be done by clicking the "Details" link next to the "license/cla" check below.

File: app/code/Magento/Catalog/Model/Product/Option/Validator/Select.php [hunk 1]
```diff
@@ -27,6 +27,22 @@ protected function checkAllValuesRemoved($values)
     }
 
     /**
+     * Unset values marked for deleting
+     *
+     * @param $values
+     * @return $this
+     */
+    protected function cleanValues(&$values)
```
Comment (vkublytskyi) [inline app/code/Magento/Catalog/Model/Product/Option/Validator/Select.php]: New method should be declared as a private if there is no strong reasons to make it protected.
Passing values by reference is strongly deprecated in Magento 2. Please rewrite method and return filtered values explicitly. 
Also more descriptive method name (e.g. `filterDeletedValues`) will increase readability of the code.

File: app/code/Magento/Catalog/Model/Product/Option/Validator/Select.php [hunk 2]
```diff
@@ -16,7 +16,7 @@ class Select extends DefaultValidator
      * @param array $values
      * @return bool
      */
-    protected function checkAllValuesRemoved($values)
+    private function checkAllValuesRemoved($values)
```
Comment (igrslv) [inline app/code/Magento/Catalog/Model/Product/Option/Validator/Select.php]: @vkublytskyi, also made this method as private

File: app/code/Magento/Catalog/Model/Product/Option/Validator/Select.php [hunk 3]
```diff
@@ -29,17 +29,17 @@ protected function checkAllValuesRemoved($values)
     /**
      * Unset values marked for deleting
      *
-     * @param $values
-     * @return $this
+     * @param array $values
+     * @return array
      */
-    protected function cleanValues(&$values)
+    private function filterDeletedValues($values)
```
Comment (igrslv) [inline app/code/Magento/Catalog/Model/Product/Option/Validator/Select.php:35]: @vkublytskyi, made it as a private method, renamed and removed reference

File: app/code/Magento/Catalog/Model/Product/Option/Validator/Select.php (hunk 2 above)
Comment (vkublytskyi) [inline app/code/Magento/Catalog/Model/Product/Option/Validator/Select.php]: We can't do this now as it's backward incompatible change.

Comment (igrslv) [inline app/code/Magento/Catalog/Model/Product/Option/Validator/Select.php]: this method is not called anywhere outside this class

Comment (igrslv) [inline app/code/Magento/Catalog/Model/Product/Option/Validator/Select.php]: anyway, will revert it back

**Comment by vkublytskyi** [comment]
@igrslv, thank you for quick response. We will proceed your PR.

File: app/code/Magento/Catalog/Model/Product/Option/Validator/Select.php (hunk 2 above)
Comment (vkublytskyi) [inline app/code/Magento/Catalog/Model/Product/Option/Validator/Select.php]: > this method is not called anywhere outside this class

But someone may extends from it and use this method.

Now inheritance is not recommended in Magento and delegation with Dependency Injection should be used. However while we have legacy code we should preserve backward compatibility. In future we will refactor such code.

**Comment by sshrewz** [comment]
Internal ticket - MAGETWO-56785

**Comment by okorshenko** [comment]
Closing this pull request. Source branch was removed

**Pull request closed without merging**